src/chronicle_storage.erl

```diff
@@ -158,25 +170,26 @@ open(PublishSnapshotFun) ->
 
     publish(Storage5).
 
-open_logs(#storage{data_dir = DataDir} = Storage) ->
-    {Orphans, Sealed, Current} = find_logs(DataDir),
-
-    NewStorage0 = Storage#storage{meta = #{},
-                                  pending_meta = #{},
-                                  config = undefined,
-                                  committed_config = undefined,
-                                  pending_configs = [],
-                                  low_seqno = ?NO_SEQNO + 1,
-                                  high_seqno = ?NO_SEQNO,
-                                  pending_high_seqno = ?NO_SEQNO,
-                                  current_snapshot = no_snapshot,
-                                  extra_snapshots = [],
-                                  log_segments = []},
-
-    %% TODO: Be more robust when dealing with corrupt log files. As long as a
-    %% consistent state can be recovered, corrupt logs should not be a fatal
-    %% error.
-    NewStorage1 =
+reset_storage(Storage) ->
```

Comment:
It's not completely clear that reset_storage/1 is useful outside of its use in open_logs. If you think it's potentially useful to keep around, then fine. Else I'd just add the function body to open_logs and remove reset_storage/1.

Reply:
I removed this fun. Plus I change the code to erase ets tables only when it's needed